godotproject/PlayerTongue.gd

```diff
@@ -58,10 +63,22 @@ func start_shoot(dirn):
 func handle_shoot(delta):
 	# Check if the raycast hit something
 	if $TongueRaycast.is_colliding():
-		start_swing($TongueRaycast.get_collider(), $TongueRaycast.get_collision_point())
+		
+		# FGTS-81 experimentation 
+		# print($TongueRaycast.get_collider().get_property_list())
+		var colliding = $TongueRaycast.get_collider()
+		
+		if colliding.get_collision_layer_bit(COLLISION_LAYER_TONGUE_CAN_DAMAGE):
+			colliding.take_damage(tongue_damage)
+			start_idle()
+		
+		elif colliding.get_collision_layer_bit(COLLISION_LAYER_TONGUEABLE):
+			start_swing($TongueRaycast.get_collider(), $TongueRaycast.get_collision_point())
+		
```

Comment:
take this out if its a part of the other PR

Reply:
removed